addons/mass_mailing/models/mass_mailing.py

```diff
@@ -40,6 +40,20 @@ def _compute_contact_nbr(self):
         mapped_data = dict([(c['list_id'][0], c['list_id_count']) for c in contacts_data])
         for mailing_list in self:
             mailing_list.contact_nbr = mapped_data.get(mailing_list.id, 0)
+    
+    @api.multi
+    def write(self, vals):
+        # Prevent archiving used mailing list
+        if 'active' in vals and not vals.get('active'):
```

Comment:
`get` check for you if the key exists in the dictionary ;) (if not, return None or the value that you given)
```suggestion
        if not vals.get('active'):
```

Reply:
It's not the same, as you want to check that you are writing `active=False`, so no way to distinguish that from an nonexistent value in the dictionary.

Reply:
Updated... if 'get' returns 'False' is because 'vals' have the key 'active' and is False.

Reply:
Well, the performance is the same, but the most important thing is that this is a backport, got as is from the original committer, so better to not change anything for seeing the same when comparing between versions. Next time, discuss this in the upstream PR :stuck_out_tongue_closed_eyes: